developers/pymavlink.md

```diff
@@ -0,0 +1,245 @@
+{'mavpackettype': 'AHRS', 'omegaIx': 0.0014122836291790009, 'omegaIy': -0.022567369043827057, 'omegaIz': 0.02394154854118824, 'accel_weight': 0.0, 'renorm_val': 0.0, 'error_rp': 0.08894175291061401, 'error_yaw': 0.0990816056728363}
+{'mavpackettype': 'HWSTATUS', 'Vcc': 4651, 'I2Cerr': 0}
+{'mavpackettype': 'SYSTEM_TIME', 'time_unix_usec': 0, 'time_boot_ms': 75191}
+{'mavpackettype': 'EKF_STATUS_REPORT', 'flags': 0, 'velocity_variance': 0.0, 'pos_horiz_variance': 0.000695356575306505, 'pos_vert_variance': 0.20162872970104218, 'compass_variance': 0.0037216085474938154, 'terrain_alt_variance': 0.04920071363449097}
+{'mavpackettype': 'VIBRATION', 'time_usec': 75191474, 'vibration_x': 0.03712763264775276, 'vibration_y': 0.03271860256791115, 'vibration_z': 0.05147671326994896, 'clipping_0': 0, 'clipping_1': 0, 'clipping_2': 0}
+{'mavpackettype': 'RAW_IMU', 'time_usec': 75430490, 'xacc': 6, 'yacc': -27, 'zacc': -1123, 'xgyro': -1, 'ygyro': 22, 'zgyro': -23, 'xmag': -353, 'ymag': -532, 'zmag': 257}
+{'mavpackettype': 'SCALED_IMU2', 'time_boot_ms': 75430, 'xacc': 39, 'yacc': 38, 'zacc': -980, 'xgyro': -45, 'ygyro': -65, 'zgyro': -13, 'xmag': 0, 'ymag': 0, 'zmag': 0}
+
+'''
+```
+
+##### Change mode
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+master = mavutil.mavlink_connection('udp:192.168.2.1:14550')
+
+# Choose a mode
+mode = 'MANUAL'
+
+# Check if mode is available
+if mode not in master.mode_mapping():
+    print('Unknown mode : {}'.format(mode))
+    print('Try:', list(master.mode_mapping().keys()))
+    return
+
+# Get mode ID
+mode_id = master.mode_mapping()[mode]
+# Set new mode
+master.set_mode(mode_id)
+```
+
+##### Arm/Disarm the vehicle
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+master = mavutil.mavlink_connection('udp:192.168.2.1:14550')
+
+# Arm
+master.arducopter_arm()
+
+# Disarm
+master.arducopter_disarm()
+```
+
+##### Send RC \(Joystick\)
+
+```py
+# Import mavutil
+from pymavlink import mavutil
+
+# Create the connection
+master = mavutil.mavlink_connection('udp:192.168.2.1:14550')
+
+# Create a function to send RC values
+# More information about Joystick channels
+# here: https://www.ardusub.com/operators-manual/rc-input-and-output.html#rc-inputs
+def set_rc_channel_pwm(id, pwm=1500):
+    """ Set RC channel pwm value
+    Args:
+        id (TYPE): Channel ID
+        pwm (int, optional): Channel pwm value 1100-1900
+    """
+    rc_channel_values = [65535 for _ in range(8)]
+    rc_channel_values[id] = pwm
+    #http://mavlink.org/messages/common#RC_CHANNELS_OVERRIDE
+    master.mav.rc_channels_override_send(
+        master.target_system,                # target_system
+        master.target_component,             # target_component
+        *rc_channel_values)                  # RC channel list, in microseconds.
+
+# Set some forward command !
+set_rc_channel_pwm(4, 1600)
+
+# Set camera tilt to 45º
+set_rc_channel_pwm(7, 1900)
```

Comment:
This isn't right [input vs output](https://www.ardusub.com/operators-manual/rc-input-and-output.html)